v2/lockrenewer.go

```diff
@@ -170,3 +182,13 @@ func (plr *peekLockRenewer) startPeriodicRenewal(ctx context.Context, message *a
 		}
 	}
 }
+
+func (plr *peekLockRenewer) renewMessageLock(ctx context.Context, message *azservicebus.ReceivedMessage, options *azservicebus.RenewMessageLockOptions) error {
+	if *plr.renewalTimeout > 0 {
+		getLogger(ctx).Info(fmt.Sprintf("renewing lock with timeout: %s", *plr.renewalTimeout))
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, *plr.renewalTimeout)
```

Comment:
Might be faster to understand here if you use `:=` with a new ctx variable name like `renewalCtx`

Reply:
using the same variable since we are only setting the timeout if the renewTimeout > 0. If it is set to a negative value, it is disabled and we don't want to mess with that?